### ISSUE TITLE
Undo broken after latest update

### ISSUE THREAD

**Issue by mister-glass** [Microsoft/vscode#64987]
Issue Type: <b>Bug</b>

After the latest update, I have to press ctrl+z multiple times to undo changes in a file. I noticed this behaviour happens when I hit Save.

VS Code version: Code 1.30.0 (c6e592b2b5770e40a98cb9c2715a8ef89aec3d74, 2018-12-11T22:29:11.253Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz (4 x 3200)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.91GB (5.02GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details>

**Comment by mjbvz** [comment]
Does this reproduce with all extensions disabled?

**Comment by mister-glass** [comment]
@mjbvz I have opened vscode with --disable-extensions arg and probably as you suspected, this bug does not reproduce. But still, the main reason that had me decide to post this issue is the fact that this started right after I updated vscode, so I still think something broke, that made one of my many extensions corrupt the undo level after I save a file.

I wanted to try and identify the problematic extension, but it turns out I can't even use the "Disable All Installed Extensions" from the Extensions panel because of this:
![image](https://user-images.githubusercontent.com/4302588/49998995-5b829400-ff9e-11e8-8bd2-34fecafc8c46.png)

I'm posting the list with all my installed extensions, maybe it helps:

<details><summary>Extensions (71)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|10.0.0
project-manager|ale|10.0.0
vscode-color|ans|0.4.5
package-to-readme|ant|0.0.3
bmgvscodepowertools|bad|1.1.2
vscode-express|Com|0.0.5
vscode-markdownlint|Dav|0.22.0
vscode-simpler-icons|dav|1.6.5
vscode-eslint|dba|1.7.2
vscode-npm-source|dku|1.0.0
gitlens|eam|9.2.0
vscode-archiver|ecm|0.0.27
EditorConfig|Edi|0.12.5
tslint|eg2|1.0.41
vscode-npm-script|eg2|0.3.5
LogFileHighlighter|emi|2.4.2
vsc-material-theme|Equ|2.6.2
vscode-todo-plus|fab|4.11.1
tortoise-svn|fan|0.1.1
auto-close-tag|for|0.5.6
code-runner|for|0.9.5
macros|ged|1.2.1
html-slim-scss-css-class-completion|gen|1.2.0
gc-excelviewer|Gra|2.1.26
asciidecorator|hel|0.2.0
randomeverything|hel|0.0.3
vscode-sort|hen|0.2.5
beautify|Hoo|1.4.7
JSDocTagComplete|Hoo|0.0.2
output-colorizer|IBM|0.1.2
docthis|joe|0.7.1
contextualduplicate|laf|0.2.0
node-module-intellisense|lei|1.5.0
expand-region|let|0.1.2
php-symbols|lin|2.1.0
ecdc|mit|0.12.0
HTMLHint|mka|0.5.0
vscode-apache|mrm|1.1.1
vscode-duplicate|mrm|1.2.1
vscode-scss|mrm|0.6.2
node-debug2|ms-|1.30.3
debugger-for-chrome|msj|4.11.1
color-highlight|nau|2.3.0
node-modules-resolve|nau|1.0.2
php-docblocker|nei|1.6.0
syncing|non|2.0.7
indent-rainbow|ode|7.2.4
require-sort|Per|0.0.1
vscode-docker|Pet|0.4.0
material-icon-theme|PKi|3.6.1
startanyshell|rem|0.3.1
vscode-sort-json|ric|1.13.0
vscode-paste-and-indent|Rub|0.0.8
partial-diff|ryu|1.4.0
yo|sam|0.9.3
MDTools|sea|1.0.1
preview-vscode|sea|2.0.3
code-settings-sync|Sha|3.2.4
vscode-nginx|sha|0.6.0
indenticator|Sir|0.6.0
vscode-multiclip|sle|0.1.5
addDocComments|ste|0.0.8
code-spell-checker|str|1.6.10
bootstrap4-vscode|the|6.0.2
selectline-statusbar|tom|0.0.2
highlight-matching-tag|vin|0.8.4
expand-selection-to-scope|vit|0.2.0
vscode-react-native|vsm|0.7.0
vscode-import-cost|wix|2.12.0
change-case|wma|1.0.0
sublime-commands|Zar|0.1.0

(6 theme extensions excluded)

</details>

**Comment by sandy081** [comment]
@mister-glass Can you please open your installed extensions folder and check if either of `ms-vscode.node-debug` or `ms-vscode.node-debug2` extensions exists in it? Use `F1 -> Open Extensions Folder` to open installed extensions folder.

**Comment by mister-glass** [comment]
@sandy081 I found ms-vscode.node-debug2-1.30.3 inside the installed extensions folder:
![image](https://user-images.githubusercontent.com/4302588/50073277-ccb18980-01e0-11e9-8e7b-ac28887c7cc5.png) 
but it's also here:
C:\Program Files\Microsoft VS Code\resources\app\extensions\ms-vscode.node-debug2
along with:
C:\Program Files\Microsoft VS Code\resources\app\extensions\ms-vscode.node-debug

**Comment by a-ignatov-parc** [comment]
I have the same issue as @mister-glass 

I've been able to pinpoint which extension exactly lead to problems with undo in my configuration. It was [`editorconfig`](https://github.com/editorconfig/editorconfig-vscode).

@sandy081 Do you need any additional info related to this problem? Or should I open issue in `editorconfig` repo?

P.S. There was no `ms-vscode.node-debug` file in my extension folder.
P.P.S. I'm running vscode under latest OSX.

**Comment by mister-glass** [comment]
@sandy081 I'm using these ones: 
https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync
https://marketplace.visualstudio.com/items?itemName=nonoroazoro.syncing

I'm not sure which one is responsible for installing node-debug in that folder

**Comment by sandy081** [comment]
@mister-glass Please check with above extensions if they are synchronising also built in extensions.

**Comment by mister-glass** [comment]
@sandy081 I managed to identify the extension that syncs node-debug2: it's this one https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync
And it doesn't even install the latest version of node-debug2... now I remember the reason why I have 2 sync extensions. Anyway, I manually deleted node-debug2 from the user extensions folder and uninstalled the code-settings-sync extension.

Also, unrelated to this issue, I went through hell and back to test this, because I needed a clean environment to install vscode so I tried it in a VirtualBox Win7 x64 image. Turns out vscode has some serious performance issues, because after many attempts to install and run the editor, I finally gave up and installed Insiders on my work pc. I'm not gonna get into details because I looked up issues related to startup performance and there are many still open.
